openedx/openedx-platform
# Drop courseware legacy view for all but Studio preview mode

This leaves most of the legacy view templating in place, because Studio's preview mode still uses it. But I've removed all other access points (casual staff access, opt-out waffle, old mongo, query param).

Test changes have taken a few forms:
* Use preview mode if the test cares about the template chrome surrounding an xblock. This requires using a staff user.
* Use the xblock_render endpoint (like the MFE does) if we just care about the xblock rendering itself (which lets the test continue using real normal users).
* Removal of the test if it specifically cares about the template chrome we are disabling but also needs normal-user mode for that chrome (like access checks or some such).

Landing this should let us close DEPR https://github.com/openedx/edx-platform/issues/35803
